### DISCUSSION TITLE
Fuse js not working on initial search

### DISCUSSION THREAD

**Discussion by studiosd** [krisk/Fuse#715] [Q&A] [answered]
I am using Fuse JS on my website that is built with Hugo. When I do a first search upon loading the page i get a single result with the variables as in the picture. I have to go to this generic result once to somehow "initialize" the search, after that Fuse JS works as expected. Every time I fully reload the page the problem persists.
Any help would be much appreciated.
![Screenshot 2023-02-21 at 18 08 37](https://user-images.githubusercontent.com/126004845/221814864-d46f3411-7237-4e65-be0c-833a42b57687.jpg)

**Comment by krisk** [comment, marked as answer]
Late reply — sorry about that.

This looks like the Fuse instance is being created before your data is fully loaded. If the list is fetched asynchronously (e.g. from a JSON file), you need to make sure the data is available before constructing the `new Fuse(list, options)`. If you create the instance with an empty or incomplete list, the first search won't work properly.

If you can't control the timing, you can use `fuse.setCollection(data)` to update the list once it's ready.
